api/src/main/java/org/apache/iceberg/RewriteJobOrder.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Locale;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Enum of supported rewrite job order, it defines the order in which the file groups should be
+ * written.
+ * <p>
+ * 1. bytes-asc: rewrite the smallest job groups first.
+ * <p>
+ * 2. bytes-desc: rewrite the largest job groups first.
+ * <p>
+ * 3. files-asc: rewrite the job groups with the least files first.
+ * <p>
+ * 4. files-desc: rewrite the job groups with the most files first.
+ * <p>
+ * 5. none: rewrite job groups in the order they were planned (no specific ordering).
+ */
+public enum RewriteJobOrder {
+  BYTES_ASC("bytes-asc"), BYTES_DESC("bytes-desc"),
+  FILES_ASC("files-asc"), FILES_DESC("files-desc"), NONE("none");
+
+  private final String orderName;
+
+  RewriteJobOrder(String orderName) {
+    this.orderName = orderName;
+  }
+
+  public String orderName() {
+    return orderName;
+  }
+
+  public static RewriteJobOrder fromName(String orderName) {
+    Preconditions.checkArgument(orderName != null, "Rewrite job order name should not be null");
+    switch (orderName.toLowerCase(Locale.ROOT)) {
```

Comment:
The case statement was for the fix below, this should use the "valueOf" logic

Please again check 
https://github.com/apache/iceberg/blob/c75ac359c1de6bf9fd4894b40009c5c42d2fee9d/api/src/main/java/org/apache/iceberg/DistributionMode.java#L52

Reply:
Made changes to use valueOf logic.

Reply:
```
  public static RewriteJobOrder fromName(String orderName) {
    Preconditions.checkArgument(orderName != null, "Rewrite job order name should not be null");
    // Replace the hyphen in order name with underscore to map to the enum value. For example: bytes-asc to BYTES_ASC
    return RewriteJobOrder.valueOf(orderName.replaceFirst("-", "_").toUpperCase(Locale.ENGLISH));
  }
```

The logic would throw `IllegalArgumentException` in case the user passes some arbitrary value. The same has been asserted in the `testInvalidOptions()` test.